src/Schema/Column/DoubleColumn.php

```diff
@@ -19,13 +23,21 @@ class DoubleColumn extends AbstractColumn
 
     public function dbTypecast(mixed $value): float|ExpressionInterface|null
     {
-        if (is_float($value)) {
-            return $value;
-        }
-
-        return match ($value) {
-            null, '' => null,
-            default => $value instanceof ExpressionInterface ? $value : (float) $value,
+        /** @var ExpressionInterface|float|null */
+        return match (gettype($value)) {
+            GettypeResult::DOUBLE => $value,
+            GettypeResult::INTEGER => (float) $value,
```

Comment:
Should it be casted to int?

Reply:
I think we should return `new Param($value, DataType::STRING)`, to avoid double type detection. But this is for another PR.